cobalt/h5vcc/h5vcc_metrics.idl

```diff
@@ -0,0 +1,36 @@
+// Copyright 2023 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Defines an interface to support interactions with Cobalt metric logging.
+// For example, a callback API such that the web client can intercept and
+// process metrics sent from Cobalt.
+
+interface H5vccMetrics {
+  // Interface for web client to bind an event handler that will be invoked
+  // every time Cobalt wants to upload a metrics payload. Only one event
+  // handler callback can be registered at any time. Duplicate calls to
+  // onMetricEvent will override any previously registered callback.
+  void onMetricEvent(H5vccMetricEventHandler eventHandler);
```

Comment:
Given this describes a new JavaScript interface to the clients, it's probably helpful to extend the comment with an example how to subscribe to it. 

In fact, we should probably add a new small HTML example under cobalt/content/demos to demonstrate the JavaScript API. That's a good thing to do even before submitting this to verify JS client API works as expected.

That example definitely needs to be provided in a design doc as well.

Reply:
Exampled added.

I'll add a demo in the same follow-up when I add tests in this same PR.